kernel/cemitcodes.ml

```diff
@@ -284,7 +284,7 @@ let emit_instr env = function
       if n <= 1 then out env (opSETFIELD0+n)
       else (out env opSETFIELD;out_int env n)
   | Ksequence _ -> invalid_arg "Cemitcodes.emit_instr"
-  | Kproj (n,p) -> out env opPROJ; out_int env n; slot_for_proj_name env p
+  | Kproj p -> out env opPROJ; out_int env (Projection.Repr.arg p); slot_for_proj_name env p
```

Comment:
Not sure if it makes sense to duplicate the int here, but I don't know what would have to be modified to avoid it.

Reply:
It does make sense, the projection name is treated abstractly by the VM.